app/src/main/java/com/example/voyagerx/MainActivity.kt

```diff
@@ -2,12 +2,41 @@ package com.example.voyagerx
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.fragment.app.Fragment
+import com.example.voyagerx.databinding.ActivityMainBinding
+import com.google.android.material.bottomnavigation.BottomNavigationView
+
 
 class MainActivity : AppCompatActivity() {
+    private lateinit var bottomNavView : BottomNavigationView
+    lateinit var binding: ActivityMainBinding
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        bottomNavView = binding.bottomNavigationView
 
-        // This is some change that i want to add to
+        val landingPageFragment = LandingPageFragment()
+        val profileFragment = ProfileFragment()
+        val settingsFragment = SettingsFragment()
+
+        changeFragmentView(landingPageFragment) //setting initial view to landing page as you only get here after bypassing login/register screens
+
+        bottomNavView.setOnItemSelectedListener {
+            when (it.itemId) {
+                R.id.landingPageFragment -> changeFragmentView(landingPageFragment)
+                R.id.profileFragment -> changeFragmentView(profileFragment)
+                R.id.settingsFragment -> changeFragmentView(settingsFragment)
+            }
+            true
+        }
     }
```

Comment:
I think you can remove the assignments and create the fragments directly. That way we're only creating fragments when we need to.

```suggestion

        changeFragmentView(LandingPageFragment()) //setting initial view to landing page as you only get here after bypassing login/register screens

        bottomNavView.setOnItemSelectedListener {
            when (it.itemId) {
                R.id.landingPageFragment -> changeFragmentView(LandingPageFragment())
                R.id.profileFragment -> changeFragmentView(ProfileFragment())
                R.id.settingsFragment -> changeFragmentView(SettingsFragment())
            }
            true
        }
    }
```

Reply:
Great suggestion

Reply:
Thanks, Celina! I implemented this change.